### PR TITLE
Add sabre docs

### PR DESCRIPTION
**Pull request by rbuysse**
Import sabre docs from master and 0-1 branches. Docs files were isolated with `git filter-branch` and then moved into a subdirectory.

### CODE REVIEW

File: docs/sabre/0.5/sabre_transaction_family.rst [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+      string creator = 3;
+    }
+
+    string name = 1;
+    repeated Version versions = 2;
+    repeated string owners = 3;
+  }
+
+ContractRegistry entries whose addresses collide are stored in a
+ContractRegsitryList.
```
Comment (danintel) [inline docs/sabre/0.5/sabre_transaction_family.rst:93]: Change ContractRegsitryList to ContractRegistryList

File: docs/sabre/0.1/sabre_cli.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+The ``--contract`` should be <contract_name:version_number>. The ``--inputs``
+and ``--outputs`` should include any namespaces or addresses that the contract
+needs to have access to. Finally the ``--payload`` should be a path to
+the file that contains the Sabre contract bytes.
+
+sabre sp
+========
+
+The ``sabre sp`` subbcommand submits a Sabre transaction that can create,
```
Comment (danintel) [inline docs/sabre/0.1/sabre_cli.rst:117]: Change subbcommand to subcommand

File: docs/sabre/0.5/application_developer_guide.rst [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+
+  use sabre_sdk::ApplyError;
+  use sabre_sdk::TransactionContext;
+  use sabre_sdk::TransactionHandler;
+  use sabre_sdk::TpProcessRequest;
+  use sabre_sdk::{WasmPtr, execute_entrypoint};
+
+- ApplyError: Used to raise an InvalidTransaction or an InternalError
+- TransactionHandler: Mimics the Sawtooth SDK TransactionHandler and should be
+    used when writting a smart contract that can also function as a Transaction
```
Comment (danintel) [inline docs/sabre/0.5/application_developer_guide.rst:50]: Change writting to writing

**Review by danintel — APPROVED** [review]
Approved after some spelling corrections

**Comment by rbuysse** [comment]
Thanks for your (very thorough) review @danintel! These first few PRs to the refresh branch are intended to move the documentation over from other repos unchanged so we can start work on converting them for use with a future version of the sawtooth website. 

I'll open some stories to track the errors you noticed.

**Review by agunde406 — APPROVED** [review]

**Pull request merged**